### PR TITLE
gen: fix mutable map generation

### PR DESCRIPTION
**Pull request by dedesite**
Key and value was not treated as pointer
Add some tests

Fixes a generation bug on FOR IN maps.

### CODE REVIEW

**Comment by dedesite** [comment]
In fact the problem was not with mutable maps, but with maps passes as function args. Maybe should I need to change the commit message and the test name ?

**Comment by dedesite** [comment]
Reformated cgen, tests should pass now.

File: TESTS.md [hunk 1]
```diff
@@ -10,6 +10,16 @@ particularly useful for checking that errors are printed.
 
 Tip: use `v -cc tcc` when compiling tests for speed.
 
+## Run only single test file
+
+In order to test only one file, use `v test` command
```
Comment (spytheman) [inline TESTS.md:15]: That is not correct.
`v test` is designed to be used for one or many _test.v files. 
It produces a summary for all of them, no matter how many you give it.
It also searches recursively in all the folders you give it too for _test.v files, so you can do `v test .` and have it run all the local tests in the current folder.

If someone wants to run a single `x_test.v` file - use just `v x_test.v` .
For statistics about the individual test functions in that file - use `v -stats x_test.v` .

**Comment by spytheman** [comment]
Other than the _test.v instructions, good work. 
I think that a separate PR for the tests may be better.

File: TESTS.md (hunk 1 above)
Comment (dedesite) [reply TESTS.md]: You are right, my explanations were not correct. What I wanted to say is that we can use `v test` on a single test file.
It was just not explicit in the doc.

**Pull request merged**